src/cli.ts

```diff
@@ -272,28 +274,23 @@ function getNameBlockchainRecord(network: CLINetworkAdapter, args: string[]) : P
 /*
  * Get all of a name's history
  */
-
-function getAllNameHistoryPages(network: CLINetworkAdapter, name: string, page: number) {
-  return new Promise((resolve, _reject) => {
-    let history = {};
-    return network.getNameHistory(name, page)
-      .then((results : any) => {
-        if (Object.keys(results).length == 0) {
-          resolve(history);
-        }
-        else {
-          history = Object.assign(history, results);
-          getAllNameHistoryPages(network, name, page + 1)
-            .then((rest : any) => {
-              history = Object.assign(history, rest);
-              resolve(history);
-            });
-        }
-      })
-      .catch((_e) => {
-        resolve(history);
-      });
-  });
+async function getAllNameHistoryPages(network: CLINetworkAdapter, name: string, page: number) {
```

Comment:
Can you test this one?  Does recursion work in async?

Reply:
Yep, tested with adding this to the top of the function:
```ts
console.log(`getAllNameHistoryPages: ${page}`);
```

Then ran `blockstack-cli get_blockchain_history judecn.id` and the output:
```
getAllNameHistoryPages: 0
getAllNameHistoryPages: 1
{
  "373622": [
    {
      "address": ....
```

Does that look right?